nmostesting/mocks/Registry.py

```diff
@@ -271,3 +427,204 @@ def heartbeat(version, node_id):
+    until = request.args.get('paging.until') or MAX_UNTIL
+    limit = min(int(request.args.get('paging.limit') or '100'), registry.paging_limit)
+
+    new_until = until
+
+    valid_resource_types = ['device', 'flow', 'node', 'receiver', 'sender', 'source', 'subscription']
+
+    resource_type = resource.rstrip("s")
+
+    if resource_type not in valid_resource_types:
+        error_message = {"code": 404,
+                         "error": "Invalid resource",
+                         "debug": resource_type}
+        return Response(json.dumps(error_message), status=404, mimetype='application/json')
+
+    registry.query_api_called = True
+
+    # Check for unsupported Query Syntax
+    for param in request.args:
+        if not param.startswith('paging') and not param == 'id':
+            abort(501)
+
+    # Check to see if resource is being requested as a query
+    if request.args.get('id'):
+        resource_id = request.args.get('id')
+        base_data.append(registry.get_resources()[resource_type][resource_id])
+    else:
+        data = registry.get_resources()[resource_type]
+
+        # only paginate for version v1.1 and up
+        if NMOSUtils.compare_api_version("v1.1", version) > 0:
+            for key, value in data.items():
+                base_data.append(value)
+        else:
+            data_list = [d for k, d in data.items()]
+            sorted_list = sorted(data_list, key=functools.cmp_to_key(compare_resources))
+
+            for value in sorted_list:
+                if NMOSUtils.compare_resource_version(value['version'], since) >= 0 \
+                        and NMOSUtils.compare_resource_version(until, value['version']) > 0:
+                    if len(base_data) < limit:
+                        base_data.append(value)
+                    else:
+                        new_until = value['version']
+                        break
+
+    response = Response(json.dumps(base_data), mimetype='application/json')
+
+    # add pagination headers for v1.1 and up
+    if NMOSUtils.compare_api_version("v1.1", version) <= 0:
+        protocol = "http"
+        host = get_default_ip()
+        port = str(registry.get_data().port)
+        if ENABLE_HTTPS:
+            protocol = "https"
+
+        # Always add a next for any resources that are added/modified after "now"
+        link = "<" + protocol + "://" + host + ":" + port \
+            + "/x-nmos/query/" + version + "/" + resource_type + "s/?paging.since=" + new_until \
+            + "&paging.limit=" + str(registry.paging_limit) + ">; rel=\"next\""
+
+        if since != MIN_SINCE:
+            link += ",<" + protocol + "://" + host + ":" + port \
+                + "/x-nmos/query/" + version + "/" + resource_type + "s/?paging.since=0:0&paging.limit=" \
+                + str(registry.paging_limit) + ">; rel=\"first\""
+
+        response.headers["Link"] = link
+        response.headers["X-Paging-Limit"] = registry.paging_limit
```

Comment:
I think you want `limit` throughout, right?

```suggestion
        link = "<" + protocol + "://" + host + ":" + port \
            + "/x-nmos/query/" + version + "/" + resource_type + "s/?paging.since=" + new_until \
            + "&paging.limit=" + str(limit) + ">; rel=\"next\""

        if since != MIN_SINCE:
            link += ",<" + protocol + "://" + host + ":" + port \
                + "/x-nmos/query/" + version + "/" + resource_type + "s/?paging.since=0:0&paging.limit=" \
                + str(limit) + ">; rel=\"first\""

        response.headers["Link"] = link
        response.headers["X-Paging-Limit"] = limit
```